### ISSUE TITLE
VSCode Intellisense error for pydantic dataclasses

### ISSUE THREAD

**Issue by jgrugru** [samuelcolvin/pydantic#4014]
I am receiving an intellisense error on an inherited pydantic dataclass. I was able to replicate this on two operating systems, one in VSCode, the other in pycharm.

You can replicate the intellisense error with this snippet:
```python
from abc import ABC, abstractmethod
from pydantic import BaseModel, dataclasses
from typing import List

# python --version -> Python 3.10.2
# pydantic=1.9.0
# Running on MacOS Monterey 12.3.1


@dataclasses.dataclass(frozen=True, eq=True)
class PersonABC(ABC):
    name: str

    @abstractmethod
    def print_bio(self):
        pass

@dataclasses.dataclass(frozen=True, eq=True)
class Jeff(PersonABC):
    age: int

    def print_bio(self):
        print(f"{self.name}: {self.age}")     # In VScode, self.name is not registered as a known variable:
                                              # `Cannot access member "name" for type "Jeff"
                                              # Member "name" is unknownPylancereportGeneralTypeIssues`

class Family(BaseModel):
    person: List[PersonABC]


jeff = Jeff(name="Jeff Gruenbaum", age=93)
print(Family(person=[jeff]))
```

The error takes place in the implemented *print_bio* function. The full error is in the comments above.

**Comment by samuelcolvin** [comment]
is this fixed by #4007?

**Comment by jgrugru** [comment]
@samuelcolvin Yes, looks like #4007 resolves this issue, closing this issue.
